File: AutoParts-BackEnd/api/src/main/java/br/com/autoparts/api/model/Pecas.java
package br.com.autoparts.api.model;


import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.Lob;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "pecas")
@Getter
@Setter
public class Pecas {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    
    private String nome;
    private String descricao;
    private Integer quantidade;
    
    @Column(name = "foto", length = 1000485760) 
    @Lob
    private byte[] foto; 
    private String marca;
    private Integer ano;
    private Double preco;
    private String modelo;
    @Column(length = 10000000)
    private String base64;

    @ManyToOne
    @JoinColumn(name = "fornecedor_id")
    private Fornecedor fornecedor;
    
    
}

File: autoparts-app/src/app/model/pecas/pecas.ts
import { Fornecedor} from "../fornecedor/fornecedor"

export class Pecas {
    pecas_id: number = 0;
    nome?: string;
    descricao?: string;
    quantidade: number = 0;
    foto?: Blob;
    base64?: string;
    marca?: string;
    ano?: number;
    preco?: number;
    modelo?: string;

    fornecedor: Fornecedor = new Fornecedor();

}
